geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -4457,9 +4466,9 @@ public QueryMonitor getQueryMonitor() {
             maxTime = FIVE_HOURS;
           }
 
-          this.queryMonitor = new QueryMonitor(this, maxTime);
-          Thread qmThread = new LoggingThread("QueryMonitor Thread", this.queryMonitor);
-          qmThread.start();
+          this.queryMonitor =
+              new QueryMonitor(() -> (ScheduledThreadPoolExecutor) Executors.newScheduledThreadPool(
+                  QueryMonitorThreadPoolSize), this, maxTime);
```

Comment:
I think this thread pool constructor has to change so that its thread(s) is a `LoggingThread`